commitizen/formats/__init__.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import sys
+
+from typing import ClassVar, Optional
+
+import importlib_metadata as metadata
+
+from commitizen.changelog import Metadata
+from commitizen.config.base_config import BaseConfig
+
+from .errors import FormatUnknown
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+FORMAT_ENTRYPOINT = "commitizen.changelog_format"
+DEFAULT_FORMAT = "markdown"
+DEFAULT_TEMPLATE_EXTENSION = "j2"
+
+
+class Format(Protocol):
+    extension: ClassVar[str]
+    """Standard known extension associated with this format"""
+
+    alternative_extensions: ClassVar[set[str]]
+    """Known alternatives extensions for this format"""
+
+    config: BaseConfig
+
+    def __init__(self, config: BaseConfig):
+        self.config = config
+
+    @property
+    def ext(self) -> str:
+        """Dotted version of extensions, as in `pathlib` and `os` modules"""
+        return f".{self.extension}"
+
+    @property
+    def template(self) -> str:
+        """Expected template name for this format"""
+        return f"CHANGELOG.{self.extension}.{DEFAULT_TEMPLATE_EXTENSION}"
+
+    @property
+    def default_changelog_file(self) -> str:
+        return f"CHANGELOG.{self.extension}"
+
+    def get_metadata(self, filepath: str) -> Metadata:
```

Comment:
Do we need to raise `Not Implemented Error`?

Reply:
Done 👍🏼